utils/ixdtf/src/parsers/records.rs

```diff
@@ -211,6 +211,26 @@ pub enum TimeDurationRecord {
         /// Seconds value.
         seconds: u64,
         /// The parsed fraction value in nanoseconds.
-        fraction: u32,
+        fraction: Fraction,
     },
 }
+
+/// A fraction value in nanoseconds or lower value.
+#[derive(Debug, Clone, Copy, PartialEq)]
+#[allow(clippy::exhaustive_enums)] // Parsed fraction must be one of the four provided options.
+pub enum Fraction {
+    /// Parsed nanoseconds value (A fraction value from 1-9 digits length)
+    Nanoseconds(u64),
+    /// Parsed picoseconds value (A fraction value from 10-12 digits length)
+    Picoseconds(u64),
+    /// Parsed femtoseconds value (A fraction value from 12-15 digits length)
+    Femtoseconds(u64),
+    /// A parsed value truncated to nanoseconds (A fraction value of 15+ digits length)
+    Truncated(u64), // An unbound fraction value truncated to nanoseconds
+}
```

Comment:
Suggestion: change this to

```rust
pub struct Fraction {
    fraction_digits: u8,
    value: u64,
}
```

and then give it functions such as

```rust
impl Fraction {
    pub fn try_to_nanoseconds(self) -> Option<u64> {
        if self.fraction_digits <= 9 {
            // compute the pow and return
        } else {
            None
        }
    }
    pub fn to_truncated_nanoseconds(self) -> u64 {
        // compute pow, either + or -
    }
}
```

Another future advantage of this model is that the Fraction can directly return a FixedDecimal, which could make fractional second formatting more efficient in icu4x.